cache.go

```diff
@@ -219,6 +219,7 @@ func (a *Cache[K, V]) Clear() {
 
```

Comment:
was thinking about just having a SetCapacity(newCap) and letting the caller do what they want. Could certainly argue it isn't appropriate for this pkg to do this and I don't think there is any barrier anymore to allowing that.

Reply:
👍 I'm good with whatever as long as we can gently step away from SetLargerCapacity.

Reply:
Ya of course, that will be slow rolled.

Added SetCapacity() (this change was only here today.. ok to break I presume, no tagged releases or anything 🤷‍♂️)
Also made a simpler SetAvailableCapacity(). The CAS loop unneeded since size isn't involved in the compare anyway.

Reply:
The trouble with SetLargerCapacity is indeed the fact that it only goes up. So if we had size=3, cap=2 and avail=0, we changed cap to 3. Now another thing could be added making, and size=4, cap=3, and on another avail call the same thing happens.

Reply:
With SetAvailableCapacity, if avail is negative we just go down so that shouldn't happen, whereas before we wouldn't because of the `if new <= cap {`. At least I think that was all that was going on. Did I miss anything?

Reply:
I think that sounds right!